src/main/java/taxi/service/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,44 @@
+package taxi.service.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls = new HashSet<>();
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession session = req.getSession();
+        Long driverId = (Long) session.getAttribute("driver_id");
+        if (driverId == null) {
+            String allowedUrl = allowedUrls.stream()
+                    .filter(i -> i.equals(req.getServletPath()))
+                    .findFirst()
+                    .orElse("/login");
```

Comment:
Using stream you loose HashSet advantege. There are only 2 elements, but...)
```suggestion
            String url = allowedUrls.contains(req.getServletPath()) ? req.getServletPath() : "/login";
```